### DISCUSSION TITLE
How to set TTL for a KEY in setUserData method?

### DISCUSSION THREAD

**Discussion by ghost** [nutgram/nutgram#252] [Q&A]
Hi. how can i set `TTL` for a `KEY` in `userCache`? (in `setUserData` method)

There is no parameter to set this ... :(

![image](https://user-images.githubusercontent.com/78247242/201632828-666385fc-98a6-4c2e-886f-2f766a9ce4fe.png)

(In here, my Nutgram's cache config is default)

I expect the method to be like this:
```php
$tg->setUserData('key', 'value', 'userId', TTL);
```
Of course, I don't know if this is true or not :)
I am a bit unfamiliar with the concept of cache.

Please help.
Thank you

**Comment by Lukasss93** [comment]
I'll implement this feature in the next update 😎

**Reply by ghost** [reply]
Thanks bro.
Can you include a method called `setBotData` and `getBotData` in the next update?

I am developing a bot creator, users send their bot tokens to my bot creator to use the services I provide them.
In this case, all the requests of the created bots are sent to only one file on the server (for example: `index.php`)

I need to put some bots settings in the cache... It would be great if there is a method other than `setUserData` for this.

For example:
```php

$bot->setBotData("KEY", "VALUE", "BOT-ID", TTL);
$bot->getBotData("KEY", ...);
$bot->deleteBotData("KEY", ...); // Delete only one key
$bot->clearBotData(); // Delete all relevant keys
// etc...
```

This feature is not only for the bot creators... it is also useful in the case that requests are sent only from one bot (we develop only one bot)... you can include settings in it. This way it will be completely separate from `setUserData`
